### DISCUSSION TITLE
Which log level is appropriate for my use case?

### DISCUSSION THREAD

**Discussion by ssoper-usgs** [FraunhoferIOSB/FROST-Server#1449] [Q&A]
Hello!

My organization is using the frost http server to serve time series data, and we are now going to update it to version 2.1.  A few years ago an internal security scan was performed on our frost-server implementation and the only issue we had was that when an error is thrown, the full error stack trace is also displayed on the error page.  My colleague looked into this issue and asked this question to see if there was a way to conditionally avoid displaying stack traces: https://github.com/FraunhoferIOSB/FROST-Server/issues/499.  We do not have the bandwidth at the moment to implement a reverse proxy, so our solution was to go into the frost code itself and remove some the e.getMessage() statements in the Service class before rebuilding the image and using it for our deployment.  We would prefer not to do this, though.

It seems like we could actually stop doing that with this update: https://github.com/FraunhoferIOSB/FROST-Server/commit/fd14468ec0a01b91330e7d398b751044882c1d1d and the addition of a variable to set the log level: https://github.com/FraunhoferIOSB/FROST-Server/blob/v2.x/CHANGELOG.md?plain=1#L11 - am I interpreting that correctly?  Is there an ideal log level to set where stack traces are not logged in the Service output?  Thanks!

**Comment by hylkevds** [comment]
The Log level settings only apply to logging, not to error messages sent to users.
Error messages sent to users should never contain stack traces or server-internal information.
Error messages in the 4xx range are user-errors, and should always contain information for the user that specifies what he did wrong.

So if you see a stack trace in service output then you need to create an issue for that so it can be fixed.
The example given in #499 is not a stack trace, and does not contain server-internal information.
